DESIGN.md

```diff
@@ -0,0 +1,30 @@
+# cloud-init Provider Design
+
+ Cloud-init is a commonly-used startup configuration utility for cloud compute instances. The cloud-init provider offers functionality to render a MIME multi-part file for use with cloud-init. Using a MIME multi-part file, the user can specify more than one type of data for cloud-init to consume. If you only have one type of data, you can leverage the built-in [`templatefile`](https://www.terraform.io/docs/configuration/functions/templatefile.html) function and a static file (like `.yml`).
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide a managed resource and data source to generate a cloud-init MIME multi-part file
```

Comment:
Is there a specification for these files which we can link to? In my brief searching, https://canonical-cloud-init.readthedocs-hosted.com/en/latest/explanation/format.html seemed the closest to existing functionality, but also not sure if we should callout that it is specifically _user data_ that this provider is intending on managing.

Reply:
I think that is the right link, i added more wording around this in the design doc as well